engine/access/rpc/backend/backend.go

```diff
@@ -394,9 +394,13 @@ func executionNodesForBlockID(
 		}
 
 		receiptCnt := len(executorIDs)
-		// if less than minExecutionNodesCnt execution receipts have been received so far, then throw an error
+		// if less than minExecutionNodesCnt execution receipts have been received so far, then return random ENs
 		if receiptCnt < minExecutionNodesCnt {
-			return flow.IdentityList{}, InsufficientExecutionReceipts{blockID: blockID, receiptCount: receiptCnt}
+			executionNodes, err := state.AtBlockID(blockID).Identities(filter.HasRole(flow.RoleExecution))
+			if err != nil {
+				return nil, fmt.Errorf("failed to retreive execution IDs for block ID %v: %w", blockID, err)
+			}
+			return executionNodes.Sample(maxExecutionNodesCnt), nil
```

Comment:
Instead of returning right away, should we still continue to the block below (`chooseExecutionNodes`, etc)? I think that function already does most of the same as what you're doing here.

Reply:
so just remove this `if`?

Reply:
No, I mean inside this `if`, just set `executorIDs` to `state.AtBlockID(blockID).Identities(filter.HasRole(flow.RoleExecution).NodeIDs()`